src/interval.jl

```diff
@@ -0,0 +1,168 @@
+    res = Arb(prec = precision(x))
+    get_abs_ubound!(midref(res), x)
+    return res
+end
+abs_ubound(x::Union{ArbOrRef,AcbOrRef}) = abs_ubound(Arf, x)
+
+"""
+    getinterval([T, ] x::Union{ArbOrRef,AcbOrRef})
+
+Returns a tuple `(l, u)` representing an interval `[l, u]` enclosing
+the ball `x`, both of them are of type `Arf`. If `T` is given convert
+to this type, supports `Arf`, `BigFloat` and `Arb`.
+
+If `x` contains `NaN` both `l` and `u` will be `NaN`.
+"""
+function getinterval(::Type{Arf}, x::ArbOrRef)
+    l, u = Arf(prec = precision(x)), Arf(prec = precision(x))
+    get_interval!(l, u, x)
+    return (l, u)
+end
+function getinterval(::Type{BigFloat}, x::ArbOrRef)
+    l, u = BigFloat(precision = precision(x)), BigFloat(precision = precision(x))
+    get_interval!(l, u, x)
+    return (l, u)
+end
+function getinterval(::Type{Arb}, x::ArbOrRef)
+    l, u = zero(x), zero(x)
+    get_interval!(midref(l), midref(u), x)
+    return (l, u)
+end
+getinterval(x::ArbOrRef) = getinterval(Arf, x)
+
+"""
+    getinterval([T, ] x::Union{ArbOrRef,AcbOrRef})
+
+Returns a tuple `(m, r)` where `m` is the midpoint of the ball and `r`
+is the radius, `m` is of type `Arf` and `r` of type `Mag`. If `T` is
+given convert both `m` and `r` to this type, supports `Arb`.
+"""
+getball(x::ArbOrRef) = (Arf(midref(x)), Mag(radref(x)))
+getball(::Type{Arb}, x::ArbOrRef) = (Arb(midref(x)), Arb(radref(x), prec = precision(x)))
+
+"""
+    union(x::ArbOrRef, y::ArbOrRef)
+    union(x::AcbOrRef, y::AcbOrRef)
+    union(x, y, z...)
+
+`union(x, y)` returns a ball containing the union of `x` and `y`.
+
+`union(x, y, z...)` returns a ball containing the union of all given
+balls.
+
+"""
+Base.union(x::ArbOrRef, y::ArbOrRef) = union!(Arb(prec = _precision((x, y))), x, y)
+Base.union(x::AcbOrRef, y::AcbOrRef) = union!(Acb(prec = _precision((x, y))), x, y)
+# TODO: Could be optimized, both for performance and enclosure
+Base.union(x::ArbOrRef, y::ArbOrRef, z::ArbOrRef, xs...) =
+    foldl(union, xs, init = union(union(x, y), z))
+Base.union(x::AcbOrRef, y::AcbOrRef, z::AcbOrRef, xs...) =
+    foldl(union, xs, init = union(union(x, y), z))
+
+"""
+    intersection(x::ArbOrRef, y::ArbOrRef)
+    intersection(x::AcbOrRef, y::AcbOrRef)
+    intersection(x, y, z...)
+
+`intersection(x, y)` returns a ball containing the intersection of `x`
+and `y`. If `x` and `y` do not overlap (as given by `overlaps(a, b)`)
+throws an `ArgumentError`.
+
+`intersection(x, y, z...)` returns a ball containing the intersection of
+all given balls. If all the balls do not overlap throws an
+`ArgumentError`.
+"""
+function intersection(x::ArbOrRef, y::ArbOrRef)
+    overlaps(x, y) ||
+        throw(ArgumentError("intersection of non-intersecting balls not allowed"))
+    res = Arb(prec = _precision((x, y)))
+    intersection!(res, x, y)
+    return res
+end
+# TODO: Could be optimized, both for performance and enclosure
+intersection(x::ArbOrRef, y::ArbOrRef, z::ArbOrRef, xs...) =
```

Comment:
IntervalArithmetic uses `Base.intersect`. maybe we could define it referring to `intersection`?
https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/e0333aee5eecc90525bd1d0b9536e984dd248ab7/src/intervals/set_operations.jl#L74